bats/tests/containers/reset.bats

```diff
@@ -0,0 +1,305 @@
+    check_docker_symlinks
+}
+
+@test 'Verify that path management was unset' {
+    check_path
+}
+
+@test 'Verify that rancher desktop context was deleted' {
+    check_rd_context
+}
+
+@test 'Verify that lima VM was deleted' {
+    check_lima
+}
+
+@test 'Verify that WSL distributions were deleted' {
+    check_WSL
+}
+
+@test 'Verify updater-longhorn.json was deleted' {
+    check_updater_longhorn_gone
+}
+
+@test 'Start Rancher Desktop 2' {
+    start_application
+}
+
+@test 'factory reset while running - keep caches' {
+    rdctl reset --factory
+}
+
+@test 'Verify that the expected directories were deleted 2' {
+    CACHE=1 check_directories
+}
+
+@test 'Verify that docker symlinks were deleted 2' {
+    check_docker_symlinks
+}
+
+@test 'Verify that path management was unset 2' {
+    check_path
+}
+
+@test 'Verify that rancher desktop context was deleted 2' {
+    check_rd_context
+}
+
+@test 'Verify that lima VM was deleted 2' {
+    check_lima
+}
+
+@test 'Verify that WSL distributions were deleted 2' {
+    check_WSL
+}
+
+@test 'Verify updater-longhorn.json was deleted 2' {
+    check_updater_longhorn_gone
+}
+
+@test 'Start Rancher Desktop 3' {
+    start_application
+}
+
+@test 'factory reset while running - delete caches' {
+    rdctl_reset --factory --cache
+}
+
+@test 'Verify that the expected directories were deleted 3' {
+    CACHE=0 check_directories
+}
+
+@test 'Verify that docker symlinks were deleted 3' {
+    check_docker_symlinks
+}
+
+@test 'Verify that path management was unset 3' {
+    check_path
+}
+
+@test 'Verify that rancher desktop context was deleted 3' {
+    check_rd_context
+}
+
+@test 'Verify that lima VM was deleted 3' {
+    check_lima
+}
+
+@test 'Verify that WSL distributions were deleted 3' {
+    check_WSL
+}
+
+@test 'Verify updater-longhorn.json was deleted when cache was retained' {
+    check_updater_longhorn_gone
+}
+
+@test 'Start up Rancher Desktop (for non-factory reset)' {
+    start_application
+}
+
+@test 'Deploy kubernetes workloads' {
+    kubectl create deployment --replicas 2 --image "${IMAGE_NGINX:?}" bats-nginx
+    kubectl wait --for=condition=Available deployment/bats-nginx
+}
+
+@test 'Make modifications to the VM' {
+    rdctl shell sudo cp /etc/os-release /etc/marker-file
+    run rdctl shell ls -l /etc/marker-file
+    assert_success
+}
+
+@test 'Reset only Kubernetes' {
+    rdctl_reset --k8s
+    wait_for_kubelet
+}
+
+@test 'Verify Kubernetes workloads removed' {
+    run kubectl get deployment/bats-nginx
+    assert_failure
+}
+
+@test 'Verify VM modifications persist' {
+    run rdctl shell ls -l /etc/marker-file
+    assert_success
+}
+
+@test 'Re-deploy kubernetes workloads' {
+    kubectl create deployment --replicas 2 --image "${IMAGE_NGINX:?}" bats-nginx
+    kubectl wait --for=condition=Available deployment/bats-nginx
+}
+
+@test 'Reset VM' {
+    rdctl_reset --vm
+}
+
+@test 'Verify VM modifications removed' {
+    run rdctl shell ls -l /etc/marker-file
```

Comment:
I think we should first check that we can run `ls` inside the VM without getting an error. Otherwise we don't know if this failure is due to the file missing, or something else:

```suggestion
    rdctl shell ls -l /etc
    run rdctl shell ls -l /etc/marker-file
```